src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -82,12 +82,26 @@ class Config:
         new.update(data)
 
         verrors = ValidationErrors()
-        if new['pool'] and not await self.middleware.call(
-            'zfs.pool.query', [('name', '=', new['pool'])]
-        ):
-            verrors.add('sysdataset_update.pool', f'Pool "{new["pool"]}" not found', errno.ENOENT)
+        if new['pool'] and new['pool'] != 'freenas-boot':
+            pool = await self.middleware.call('pool.query', [['name', '=', new['pool']]])
+            if not pool:
+                verrors.add(
+                    'sysdataset_update.pool',
+                    f'Pool "{new["pool"]}" not found',
+                    errno.ENOENT
+                )
+            elif pool[0]['encrypt'] == 2:
+                # This will cover two cases - passphrase being set for a pool and that it might be locked as well
+                verrors.add(
+                    'sysdataset_update.pool',
+                    f'Pool "{new["pool"]}" has a passphrase set, please select another pool for system dataset'
```

Comment:
It would be clearer to say that "The system dataset cannot be placed on an encrypted pool.", but I don't know if that is true, or whether we just don't allow the system dataset on encrypted pools if they have a passphrase.
```suggestion
                    f'Pool "{new["pool"]}" has an encryption passphrase set. The system dataset cannot be placed on this pool.'
```

Reply:
`we just don't allow the system dataset on encrypted pools if they have a passphrase` is accurate. I think the change outlined by you in this case is good. I'm marking this as resolved with committing the suggestion, please let me know otherwise